esp32/machine_pin.c

```diff
@@ -335,6 +342,22 @@ STATIC mp_obj_t machine_pin_irq_trigger(size_t n_args, const mp_obj_t *args) {
 }
 STATIC MP_DEFINE_CONST_FUN_OBJ_VAR_BETWEEN(machine_pin_irq_trigger_obj, 1, 2, machine_pin_irq_trigger);
 
+STATIC mp_uint_t pin_ioctl(mp_obj_t self_in, mp_uint_t request, uintptr_t arg, int *errcode) {
```

Comment:
Please move this function to earlier in the file so the forward declaration of it is not needed (put the function just before machine_pin_locals_dict_table).